pkg/trustedresources/verify.go

```diff
@@ -100,18 +91,81 @@ func VerifyPipeline(ctx context.Context, pipelineObj v1beta1.PipelineObject, k8s
 		ObjectMeta: pm,
 		Spec:       pipelineObj.PipelineSpec(),
 	}
-	verifiers, err := getVerifiers(ctx, k8s)
+
+	return verifyResource(ctx, &pipeline, k8s, signature, source, verificationPolicies)
+}
+
+// verifyResource verifies a resource
+func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.Interface, signature []byte, source string, verificationPolicies []*v1alpha1.VerificationPolicy) error {
+	if len(verificationPolicies) != 0 {
+		// get all matching policies
+		matchedPolicies := []*v1alpha1.VerificationPolicy{}
+		for _, vp := range verificationPolicies {
+			match := false
+			for _, ra := range vp.Spec.Resources {
+				matching, _ := regexp.MatchString(ra.Pattern, source)
+				// If not matching then skip fetching keys
+				if matching {
+					match = true
+					break
+				}
+			}
+			if match {
+				matchedPolicies = append(matchedPolicies, vp)
+			}
+		}
+		// verify the resource on matchedPolicies
+		verified := 0
+		for _, mp := range matchedPolicies {
+			verifiers, err := getVerifiersFromPolicy(ctx, k8s, source, mp)
+			if err != nil {
+				return err
+			}
+			for _, verifier := range verifiers {
+				// if one of the verifier passes verification, then this policy passes verification
+				if err := verifyInterface(resource, verifier, signature); err == nil {
+					verified++
+					break
+				}
+			}
+		}
+		if len(matchedPolicies) == 0 {
+			return fmt.Errorf("No matching policies found for resource %s from source %s", resource.GetName(), source)
+		}
+		// make sure all matched policies pass verification
+		if verified == len(matchedPolicies) {
+			return nil
+		}
+		return fmt.Errorf("resource %s in namespace %s fails verification", resource.GetName(), resource.GetNamespace())
+	}
+
+	verifiers, err := getVerifiersFromConfigMap(ctx, k8s)
```

Comment:
How are these verifiers different from the verifiers checked above? Should these be combined?

Reply:
Those are from configmap which we introduced before. @bobcatfish suggested we removed the configmap way. I think it makes sense if we consider to remove it?
Since the current verificationpolicy has better user experience and more features support.